File: onlinepayments-sdk/src/main/java/com/onlinepayments/sdk/client/android/model/PreparedPaymentRequest.java
/*
 * Copyright 2020 Global Collect Services B.V
 */

package com.onlinepayments.sdk.client.android.model;

import java.security.InvalidParameterException;

/**
 * Contains all encrypted paymentrequest data needed for doing a payment.
 */
public class PreparedPaymentRequest {

	private String encryptedFields;
	private String encodedClientMetaInfo;

	/**
	 * @deprecated In a future release, this constructor will become internal to the SDK.
	 */
	@Deprecated
	public PreparedPaymentRequest(String encryptedFields, String encodedClientMetaInfo) {

		if (encryptedFields == null) {
			throw new InvalidParameterException("Error creating PreparedPaymentRequest, encryptedFields may not be null");
		}
		if (encodedClientMetaInfo == null) {
			throw new InvalidParameterException("Error creating PreparedPaymentRequest, encodedClientMetaInfo may not be null");
		}

		this.encryptedFields = encryptedFields;
		this.encodedClientMetaInfo = encodedClientMetaInfo;
	}

	public String getEncodedClientMetaInfo() {
		return encodedClientMetaInfo;
	}

	public String getEncryptedFields() {
		return encryptedFields;
	}
}

File: sync/Worldline/repo/onlinepayments-sdk/src/main/kotlin/com/onlinepayments/sdk/client/android/model/PreparedPaymentRequest.kt
/*
 * Do not remove or alter the notices in this preamble.
 *
 * Copyright © 2025 Worldline and/or its affiliates.
 *
 * All rights reserved. License grant and user rights and obligations according to the applicable license agreement.
 *
 * Please contact Worldline for questions regarding license and user rights.
 */
package com.onlinepayments.sdk.client.android.model

/**
 * Contains all encrypted payment request data needed for doing a payment.
 */
data class PreparedPaymentRequest(
    val encryptedFields: String,
    val encodedClientMetaInfo: String
)
